src/tabnote.ts

```diff
@@ -228,21 +240,21 @@ export class TabNote extends StemmableNote {
   }
 
   // Set the `stave` to the note
-  setStave(stave) {
+  setStave(stave: Stave): this {
     super.setStave(stave);
-    this.context = stave.context;
+    this.setContext(stave.getContext());
 
     // Calculate the fret number width based on font used
     let i;
-    if (this.context) {
-      const ctx = this.context;
+    const ctx = this.getContext();
```

Comment:
You can remove a `getContext` call with:

```typescript
setStave(..) {
  super.setStave(stave);
  const ctx = stave.getContext();
  this.setContext(ctx);

  let i;
  ...
}
```

Reply:
done